packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -441,75 +539,76 @@ const start = async (
+    return add(roundId, 1) === rrId && rounds.get(rrId).updatedAt === 0n;
   };
 
   /**
-   * @param {OracleKey} _oracle
-   * @param {bigint} _roundId
+   * @param {string} oracleAddr
+   * @param {bigint} roundId
    * @param {Timestamp} blockTimestamp
+   * @returns {string?} error message, if there is one
    */
-  const validateOracleRound = (_oracle, _roundId, blockTimestamp) => {
+  const validateOracleRound = (oracleAddr, roundId, blockTimestamp) => {
     // cache storage reads
-    const startingRound = oracleStatuses.get(_oracle).startingRound;
+    const startingRound = oracleStatuses.get(oracleAddr).startingRound;
     const rrId = reportingRoundId;
 
     let canSupersede = true;
-    if (_roundId > 1n) {
-      canSupersede = supersedable(subtract(_roundId, 1), blockTimestamp);
+    if (roundId > 1n) {
+      canSupersede = supersedable(subtract(roundId, 1), blockTimestamp);
     }
 
     if (startingRound === 0n) return 'not enabled oracle';
-    if (startingRound > _roundId) return 'not yet enabled oracle';
-    if (oracleStatuses.get(_oracle).endingRound < _roundId)
+    if (startingRound > roundId) return 'not yet enabled oracle';
+    if (oracleStatuses.get(oracleAddr).endingRound < roundId)
       return 'no longer allowed oracle';
-    if (oracleStatuses.get(_oracle).lastReportedRound >= _roundId)
+    if (oracleStatuses.get(oracleAddr).lastReportedRound >= roundId)
       return 'cannot report on previous rounds';
     if (
-      _roundId !== rrId &&
-      _roundId !== add(rrId, 1) &&
-      !previousAndCurrentUnanswered(_roundId, rrId)
+      roundId !== rrId &&
+      roundId !== add(rrId, 1) &&
+      !previousAndCurrentUnanswered(roundId, rrId)
     )
       return 'invalid round to report';
-    if (_roundId !== 1n && !canSupersede)
+    if (roundId !== 1n && !canSupersede)
       return 'previous round not supersedable';
-    return '';
+    return null;
   };
 
   /**
-   * @param {OracleKey} _oracle
-   * @param {bigint} _roundId
+   * @param {string} oracleAddr
+   * @param {bigint} roundId
    */
-  const delayed = (_oracle, _roundId) => {
-    const lastStarted = oracleStatuses.get(_oracle).lastStartedRound;
-    return _roundId > add(lastStarted, restartDelay) || lastStarted === 0n;
+  const delayed = (oracleAddr, roundId) => {
+    const lastStarted = oracleStatuses.get(oracleAddr).lastStartedRound;
+    return roundId > add(lastStarted, restartDelay) || lastStarted === 0n;
   };
 
   /**
    * a method to provide all current info oracleStatuses need. Intended only
    * only to be callable by oracleStatuses. Not for use by contracts to read state.
    *
-   * @param {OracleKey} _oracle
+   * @param {string} oracleAddr
    * @param {Timestamp} blockTimestamp
    */
-  const oracleRoundStateSuggestRound = (_oracle, blockTimestamp) => {
-    const oracle = oracleStatuses.get(_oracle);
+  const oracleRoundStateSuggestRound = (oracleAddr, blockTimestamp) => {
```

Comment:
I wonder what an `oracleAddr` is... I _think_ it's: an address to which an invitation to this contract was sent. Why would this contract know the addresses? hm.

Reply:
> I wonder what an oracleAddr is... I think it's: an address to which an invitation to this contract was sent.

Correct. I'll make clearer by documentation.

> Why would this contract know the addresses?

It needs to know the oracles and it's keyed on their public address. What's the concern?

Reply:
The fact that the `.instance` test works suggests that the keys _don't_ need to be addresses of the validators; the validators could be identified as 'A', 'B', and 'C' and the contract would work fine, yes? The convention of using addresses seems to be external to the contract.

Reply:
Oh yes, good point. I'll change it to be oracleKey.